homeassistant/components/sharkiq/vacuum.py

```diff
@@ -191,6 +198,24 @@ async def async_locate(self, **kwargs: Any) -> None:
         """Cause the device to generate a loud chirp."""
         await self.sharkiq.async_find_device()
 
+    async def async_clean_room(self, rooms: list[str], **kwargs: Any) -> None:
+        """Clean specific rooms."""
+        rooms_to_clean = []
+        valid_rooms = self.available_rooms or []
+        for room in rooms:
+            if room in valid_rooms:
+                rooms_to_clean.append(room)
+            else:
+                raise ServiceValidationError(
+                    translation_domain=DOMAIN,
+                    translation_key="invalid_room",
+                    translation_placeholders={"room": room},
+                )
```

Comment:
Ok, I went with just updating the error message, mainly because the list of rooms didn't print right... and I really think just listing out the first invalid room is fine.